middleware/http/wasm/httpwasm.go

```diff
@@ -0,0 +1,161 @@
+// NewMiddleware returns a new wasm basic middleware.
+func NewMiddleware(logger logger.Logger) dapr.Middleware {
+	return &middleware{logger: logger}
+}
+
+// GetHandler returns the HTTP handler provided by wasm basic middleware.
+func (m *middleware) GetHandler(metadata dapr.Metadata) (func(next http.Handler) http.Handler, error) {
+	rh, err := m.getHandler(metadata)
+	if err != nil {
+		return nil, err
+	}
+	return rh.requestHandler, nil
+}
+
+// getHandler is extracted for unit testing.
+func (m *middleware) getHandler(metadata dapr.Metadata) (*requestHandler, error) {
+	meta, err := m.getMetadata(metadata)
+	if err != nil {
+		return nil, fmt.Errorf("wasm basic: failed to parse metadata: %w", err)
+	}
+
+	var stdout, stderr bytes.Buffer
+	mw, err := wasmnethttp.NewMiddleware(ctx, meta.guest,
+		handler.Logger(m),
+		handler.ModuleConfig(wazero.NewModuleConfig().
+			WithStdout(&stdout). // reset per request
+			WithStderr(&stderr). // reset per request
+			// The below violate sand-boxing, but allow code to behave as expected.
+			WithRandSource(rand.Reader).
+			WithSysNanosleep().
+			WithSysWalltime().
+			WithSysNanosleep()))
+	if err != nil {
+		return nil, err
+	}
+
+	return &requestHandler{mw: mw, logger: m.logger, stdout: &stdout, stderr: &stderr}, nil
+}
+
+// IsEnabled implements the same method as documented on api.Logger.
+func (m *middleware) IsEnabled(l api.LogLevel) bool {
+	// TODO: https://github.com/dapr/kit/pull/24
+	// return m.logger.IsOutputLevelEnabled(toDaprLevel(l))
+	return true
+}
+
+// Log implements the same method as documented on api.Logger.
+func (m *middleware) Log(_ context.Context, level api.LogLevel, message string) {
+	switch level {
+	case api.LogLevelError:
+		m.logger.Error(message)
+	case api.LogLevelWarn:
+		m.logger.Warn(message)
+	case api.LogLevelInfo:
+		m.logger.Info(message)
+	case api.LogLevelDebug:
+		m.logger.Debug(message)
+	default: // same as api.LogLevelNone
+		return
+	}
+}
+
+func (m *middleware) getMetadata(metadata dapr.Metadata) (*middlewareMetadata, error) {
+	b, err := json.Marshal(metadata.Properties)
+	if err != nil {
+		return nil, err
+	}
+
+	var data middlewareMetadata
+	err = json.Unmarshal(b, &data)
+	if err != nil {
+		return nil, err
+	}
+
+	if data.Path == "" {
+		return nil, errors.New("missing path")
+	}
+
+	data.guest, err = os.ReadFile(data.Path)
+	if err != nil {
+		return nil, fmt.Errorf("error reading path: %w", err)
+	}
+
+	return &data, nil
+}
+
+type requestHandler struct {
+	mw             wasmnethttp.Middleware
+	logger         logger.Logger
+	stdout, stderr *bytes.Buffer
+}
+
+func (rh *requestHandler) requestHandler(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		h := rh.mw.NewHandler(r.Context(), next)
+		defer func() {
+			rh.stdout.Reset()
+			rh.stderr.Reset()
+		}()
+
+		h.ServeHTTP(w, r)
+
+		if stdout := rh.stdout.String(); len(stdout) > 0 {
+			rh.logger.Debugf("wasm stdout: %s", stdout)
```

Comment:
Is the log level always DEBUG or sync with Dapr itself?

Reply:
Hi @daixiang0 - the SDK includes a `log` method that is synced with Dapr

https://github.com/http-wasm/http-wasm-guest-tinygo/blob/main/handler/api/api.go#L48
https://github.com/dapr/components-contrib/pull/2239/files/d54eb0d1c97f1a6c2b0e1bfd320eab01ee10e479#diff-90c834bbe0d45903a30fffa3a23ebed48a983e5e9b32868d5b513d8c09dcad8dR93

However the middleware may write directly to `stdout`/`stderr` without using that function, which is what is logged here. We need to pick a level for logging these as the middleware didn't.

Does `Debug` look OK for that?

Reply:
I know, it is ok.